packages/nextcloud/doc/development.md

```diff
@@ -0,0 +1,41 @@
+# Development
+
+Except for WebDAV all client code is generated using OpenAPI specifications which can be found in the `lib/src/api/` folder.  
+These OpenAPI specifications are [generated](https://github.com/nextcloud/openapi-extractor) from the PHP source code.
+
+## Checking test do not interfere with each other
+
+All tests should be standalone and never affect any other tests.
+With enough caution this is feasible to implement, but to verify the tests do not depend on the order of execution they should be run multiple times with different seeds:
+
+```sh
+for i in {0..4}; do fvm flutter test --test-randomize-ordering-seed $i; done
+```
+
+## Testing against development versions
+
+Make sure to have the following apps in your development setup and the appropriate versions you want to test checked out:
```

Comment:
> in your development setup

What do you mean by that?
That you need the git submodules checked out?

Reply:
The setup when you develop on the server or an app for server. I'll clarify this